cupy/linalg/norms.py

```diff
@@ -39,12 +38,12 @@ def norm(x, ord=None, axis=None, keepdims=False):
         ndim = x.ndim
         if (ord is None or (ndim == 1 and ord == 2) or
                 (ndim == 2 and ord in ('f', 'fro'))):
-            if issubclass(x.dtype.type, numpy.complexfloating):
+            if x.dtype.kind == 'c':
                 s = abs(x.ravel())
                 s *= s
                 ret = cupy.sqrt(s.sum())
             else:
-                ret = cupy.sqrt((x.ravel() ** 2).sum())
+                ret = cupy.sqrt((x * x).ravel().sum())
```

Comment:
`ravel` can be removed here. It'll just return a flattened view of `x * x`. As `x * x` will always be contiguous in memory we won't rejoice any performance gains by raveling here since `sum` (or any other reduction in CuPy) will squash contiguous dimensions before performing the computations.
```suggestion
                ret = cupy.sqrt((x * x).sum())
```

Reply:
right, the ravel can definitely be removed